realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -527,7 +532,7 @@ private void completedAsyncRealmObject(QueryUpdateTask.Result result) {
                     if (rowPointer != 0 && emptyAsyncRealmObject.containsKey(realmObjectWeakReference)) {
                         // cleanup a previously empty async RealmObject
                         emptyAsyncRealmObject.remove(realmObjectWeakReference);
-                        realmObjects.put(realmObjectWeakReference, null);
+                        realmObjects.put(realmObjectWeakReference, Boolean.TRUE);
```

Comment:
Since `Boolean.TRUE` is used twice I would move it to a properly named class constant